common/src/main/scala/com/gu/media/youtube/YouTubePartnerApi.scala

```diff
@@ -144,38 +144,29 @@ trait YouTubePartnerApi { this: YouTubeAccess with Logging =>
     }
   }
 
-  private def getPagedVideoClaims(request: YouTubePartner#ClaimSearch#List, items: List[ClaimSnippet]):
-    List[ClaimSnippet] = {
-
-    val response = request.execute()
-
-    val allItems = if (response.getPageInfo.getTotalResults == 0) items
-      else items ++ response.getItems.asScala.toList
-
-    response.getNextPageToken match {
-      case null => allItems
-      case token => {
-        val nextRequest = request.setPageToken(token)
-        getPagedVideoClaims(nextRequest, allItems)
-      }
-    }
-  }
-
-  def createOrUpdateClaim(atomId: String, videoId: String, blockAds: Boolean): Either[VideoUpdateError, String] = {
-    val request: YouTubePartner#ClaimSearch#List = partnerClient
+  private def getPartnerClaim(videoId: String): Option[ClaimSnippet] = {
+    val response = partnerClient
       .claimSearch()
       .list
       .setVideoId(videoId)
+      .setOnBehalfOfContentOwner(contentOwner)
+      .setIncludeThirdPartyClaims(false)
+      .setPartnerUploaded(true)
+      .execute()
+
+    if (response.getPageInfo.getTotalResults == 0) {
+      None
+    } else {
+      response.getItems.asScala.toList.headOption
+    }
+  }
 
+  def createOrUpdateClaim(atomId: String, videoId: String, blockAds: Boolean): Either[VideoUpdateError, String] = {
     try {
-
-      val videoClaims = getPagedVideoClaims(request, List())
-
-      videoClaims.find(claimSnippet => !claimSnippet.getThirdPartyClaim) match {
+      getPartnerClaim(videoId) match {
         case Some(claimSnippet) => {
```

Comment:
Limited Scala knowledge alert: can this be done by directly mapping on the result of type Option?

Reply:
we're doing that here:

```scala
getPartnerClaim(videoId) match { //getPartnerClaim(videoId) is type Option[ClaimSnippet]
  case Some(claimSnippet) => ...
  case None => ...
}
```

unless I've misunderstood the question?

Reply:
got it, there's a different path on none :)

Reply:
discussed IRL: we want to do something on the unhappy path, so a `map` wouldn't work